go-controller/pkg/metrics/recorder/recorder.go

```diff
@@ -0,0 +1,494 @@
+	delete(pm.records, podUID)
+}
+
+func (pm *Pod) AddLSP(podUID kapimtypes.UID) {
+	if pm.queue != nil {
+		pm.queue.Add(pItem{op: addLogicalSwitchPort, podUID: podUID, timestamp: time.Now()})
+	}
+}
+
+func (pm *Pod) addLSP(podUID kapimtypes.UID, t time.Time) {
+	var r *podRecord
+	if r = pm.getRecord(podUID); r == nil {
+		klog.V(5).Infof("Add Logical Switch Port event expected pod with UID %q in cache", podUID)
+		return
+	}
+	if r.timestampType != firstSeen {
+		klog.V(5).Infof("Unexpected last event type (%d) in cache for pod with UID %q", r.timestampType, podUID)
+		return
+	}
+	metricFirstSeenLSPLatency.Observe(t.Sub(r.timestamp).Seconds())
+	r.timestamp = t
+	r.timestampType = logicalSwitchPort
+}
+
+func (pm *Pod) addPortBinding(m model.Model, t time.Time) {
+	var r *podRecord
+	row := m.(*sbdb.PortBinding)
+	podUID := getPodUIDFromPortBinding(row)
+	if podUID == "" {
+		return
+	}
+	if r = pm.getRecord(podUID); r == nil {
+		klog.V(5).Infof("Add port binding event expected pod with UID %q in cache", podUID)
+		return
+	}
+	if r.timestampType != logicalSwitchPort {
+		klog.V(5).Infof("Unexpected last event entry (%d) in cache for pod with UID %q", r.timestampType, podUID)
+		return
+	}
+	metricLSPPortBindingLatency.Observe(t.Sub(r.timestamp).Seconds())
+	r.timestamp = t
+	r.timestampType = portBinding
+}
+
+func (pm *Pod) updatePortBinding(old, new model.Model, t time.Time) {
+	var r *podRecord
+	oldRow := old.(*sbdb.PortBinding)
+	newRow := new.(*sbdb.PortBinding)
+	podUID := getPodUIDFromPortBinding(newRow)
+	if podUID == "" {
+		return
+	}
+	if r = pm.getRecord(podUID); r == nil {
+		klog.V(5).Infof("Port binding update expected pod with UID %q in cache", podUID)
+		return
+	}
+	if oldRow.Chassis == nil && newRow.Chassis != nil && r.timestampType == portBinding {
+		metricPortBindingChassisLatency.Observe(t.Sub(r.timestamp).Seconds())
+		r.timestamp = t
+		r.timestampType = portBindingChassis
+	}
+	if oldRow.Up != nil && !*oldRow.Up && newRow.Up != nil && *newRow.Up && r.timestampType == portBindingChassis {
+		metricPortBindingUpLatency.Observe(t.Sub(r.timestamp).Seconds())
+	}
+}
+
+// getRecord returns record from map with func argument as the key
+func (pm *Pod) getRecord(podUID kapimtypes.UID) *podRecord {
+	r, ok := pm.records[podUID]
+	if !ok {
+		klog.V(5).Infof("Cache entry expected pod with UID %q but failed to find it", podUID)
+		return nil
+	}
+	return r
+}
+
+func getPodUIDFromPortBinding(row *sbdb.PortBinding) kapimtypes.UID {
+	if isPod, ok := row.ExternalIDs["pod"]; !ok || isPod != "true" {
+		return ""
+	}
+	podUID, ok := row.Options["iface-id-ver"]
+	if !ok {
+		return ""
+	}
+	return kapimtypes.UID(podUID)
+}
+
+func (cm *Config) run(nbClient libovsdbclient.Client, kube kube.Interface, stop <-chan struct{}) {
+	// Override command line to disable this functionality if cluster node size is large to protect SB DB
+	if isNodeCountGreaterThan(kube, configDurationNodeLimit) {
```

Comment:
I am reconsidering this from SLO pov.
Compare with the `cpr.Pod.run(sbClient, stop)`, how much more load will be generated by this?

Reply:
And since we have the flag `ovnkube_config_duration_monitor_enable` to enable/disable this.
Should we just leave the flag as a single switch?

Reply:
> I am reconsidering this from SLO pov.
Compare with the cpr.Pod.run(sbClient, stop), how much more load will be generated by this?

From OVNs POV, very little. It must send out events to ovnkube-master for port_binding table.

>And since we have the flag ovnkube_config_duration_monitor_enable to enable/disable this.
Should we just leave the flag as a single switch?

Id rather have separate switches but since I didnt add a switch originally, I am not introducing one in this PR.